controllers/fenceagentsremediation_controller.go

```diff
@@ -202,6 +213,7 @@ func (r *FenceAgentsRemediationReconciler) Reconcile(ctx context.Context, req ct
 		cmd := append([]string{far.Spec.Agent}, faParams...)
 		// The Fence Agent is excutable and the parameters structure are valid, but we don't check their values
 		r.Log.Info("Execute the fence agent", "Fence Agent", far.Spec.Agent, "Node Name", req.Name)
+		eventGenerator(r.Recorder, far, v1alpha1.EventReasonFenceAgentExecuted)
```

Comment:
I think the whole eventGenerator -> eventCreator is an over complication.
IMO they can be removed since they can be easily replaced by a clear one liner:
For example in this case:
instead of `eventGenerator(r.Recorder, far, v1alpha1.EventReasonFenceAgentExecuted)`
Just use: `r.Recorder.Event(far, v1alpha1.EventTypeNormal, v1alpha1.EventReasonFenceAgentExecuted, v1alpha1.EventMessageFenceAgentExecuted)`

Reply:
>IMO they can be removed since they can be easily replaced by a clear one liner:

I agree it is complicated on the one hand but on the other hand, the one-liner seems long and less readable. I might prefer the one-liner eventually.